### PR TITLE
added GPGME::Key.valid? method

### PR DESCRIPTION
**Pull request by krtschmr**
would add functionality for https://github.com/ueno/ruby-gpgme/issues/131

### CODE REVIEW

**Comment by dansketcher** [comment]
Hi @krtschmr, would you be able to add a test or two for this one? Helps keep the coverage up :)

Cheers
Dan

**Comment by krtschmr** [comment]
@dansketcher i've added a test for true and false. i bumped the version by a minor. if you don't like that let me know and i'll remove that commit so you can bump the version. thanks for merging.

**Comment by krtschmr** [comment]
@dansketcher can we please merge this?

**Comment by dansketcher** [comment]
Yeah drop out the version bump there, it needs to be done elsewhere as well, with the docs too.
We can merge, just the day job keeping me busy ;)

**Comment by krtschmr** [comment]
@dansketcher added quick docs and removed the bump. thanks

**Comment by alexjfisher** [comment]
This doesn't just check whether a key is valid, but it also imports it if it is.  If run on a block containing multiple keys, it will return false but have also imported *all* of the keys.

**Comment by krtschmr** [comment]
@alexjfisher urgh - now that you say it, makes sense 🤦🏿 
do you happen to have an idea how to then just check a key?

**Comment by krtschmr** [comment]
> This doesn't just check whether a key is valid, but it also imports it if it is. If run on a block containing multiple keys, it will return false but have also imported _all_ of the keys.

is this still a problem?

File: lib/gpgme/key.rb [hunk 1]
```diff
@@ -122,6 +122,12 @@ def import(keydata, options = {})
           ctx.import_result
         end
       end
+
+      # Checks if a key is valid
+      def valid?(key)
+        GPGME::Key.import(key).considered == 1
+      end
```
Comment (krtschmr) [inline lib/gpgme/key.rb:129]: @alexjfisher 

one idea could be to check if that key is already imported. 
if not, we deleted it afterwards again.
it's a bit ugly, but at least the keyring stays clean. would that be a sufficient effort?

**Pull request merged**